paddle/fluid/inference/tensorrt/convert/conv2d_op.cc

```diff
@@ -35,12 +35,20 @@ class Conv2dOpConverter : public OpConverter {
     auto* Y_v = scope.FindVar(op_desc.Input("Filter").front());
     PADDLE_ENFORCE_NOT_NULL(Y_v);
     auto* Y_t = Y_v->GetMutable<framework::LoDTensor>();
-    auto* weight_data = Y_t->mutable_data<float>(platform::CPUPlace());
 
-    PADDLE_ENFORCE_EQ(Y_t->dims().size(), 4UL);
-    const int n_output = Y_t->dims()[0];
-    const int filter_h = Y_t->dims()[2];
-    const int filter_w = Y_t->dims()[3];
+    platform::CPUPlace cpu_place;
+    framework::LoDTensor* weight_tensor = new framework::LoDTensor();
```

Comment:
wrap this raw pointer to a unique_ptr when it creates.

```
std::unique_ptr<framework::LoDTensor*> weight_tensor(new frameowrk::LoDTensor();
```

Reply:
I use the unique_ptr  ```engine_->weight_map[op_desc.Input("Y").front()] =
        std::move(std::unique_ptr<framework::Tensor>(weight_tensor));```